### DISCUSSION TITLE
[Call to Collaborate] Petite Vue is amazing. But it needs work.

### DISCUSSION THREAD

**Discussion by jikkuatwork** [vuejs/petite-vue#235] [General]
I discovered Petite Vue a few days ago and was captivated by its simplicity. Finally, it feels like something I can work with. I'm done with the chaos of `node_modules`. My goal is to build simple apps with great developer experience and simplicity.

I've been tinkering with it but having trouble with reactive updates in nested components. I've shared a [minimal example on CodePen](https://codepen.io/jikkuatwork/pen/WNBeLLV?editors=1010), highlighting common issues. This could serve as a starting point for discussing quirks and challenges in the framework. Feel free to fork and experiment with it.

Here's what I'm exploring with this project:

| Feature        | Description                               | Status                             | Remarks                               |
|----------------|-------------------------------------------|-------------------------------------|---------------------------------------|
| Prop Drilling  | Passing parameters to deeper components  | Working                            | As expected                          |
| Reactivity     | Auto re-render on data changes           | Works at the root level            | Nested components don't update       |
| Errors         | Framework error reporting                 | Largely absent                     | Hard to debug complex issues         |
| Tooling        | Development tools                         | Mostly absent                      | Haven't tried Vue tools or explored deeply |

Note: If anybody is interested, I would like to form a community around it. Please let me know and I would love to help & support its growth.

Questions/Discussion points:
1. I am forcing rerender by removing all nested components and repopulating them; is this the way?
2. State in my deeper components are lost on this hack; if else, I will need to keep all my data in the root
3. Can anybody contribute examples for `v-effect`, `reactive`, `$nextTick`
4. Can anybody review my CodePen, am I doing it right? I feel I am jumping around with hacks in reactivity (may be its my ignorance)

**Comment by fnick851** [comment]
I like petite-vue over Alpine.js too. For example, for loop doesn't require the `<template />` tag. When used for elements in SVG, `<template>` tag wouldn't work, which is a big deal breaker for me.

**Reply by jikkuatwork** [reply]
Yes, and also it can easily be upgraded to Vue if need be.
